numpy/testing/utils.py

```diff
@@ -1937,3 +1968,231 @@ def __exit__(self, *exc_info):
                 mod.__warningregistry__.clear()
             if mod in self._warnreg_copies:
                 mod.__warningregistry__.update(self._warnreg_copies[mod])
+
+
+class suppressed_warnings(object):
```

Comment:
``suppress_warnings`` is already slightly better I think (mirrors ``catch_warnings`` better). Can't think of anything much better than that.